lib/sqlitex/query.ex

```diff
@@ -0,0 +1,88 @@
+defmodule Sqlitex.Query do
+  use Pipe
+
+  defstruct bindings: [],
+            column_names: [],
+            column_types: [],
+            database: nil,
+            into: [],
+            raw_data: [],
+            sql: nil,
+            statement: nil
+
+  def query(db, sql, opts \\ []) do
+    pipe_matching {:ok, %Sqlitex.Query{}},
+      {:ok, %Sqlitex.Query{bindings: bindings_from_opts(opts), into: into_from_opts(opts), database: db, sql: sql}}
```

Comment:
`Sqlitex.Query` in all these struct references can be replaced with `__MODULE__`.  This makes it easier to deal with if you ever need to rename things.